pandas/core/indexes/datetimes.py

```diff
@@ -36,7 +32,16 @@ def _new_DatetimeIndex(cls, d):
     if "data" in d and not isinstance(d["data"], DatetimeIndex):
         # Avoid need to verify integrity by calling simple_new directly
         data = d.pop("data")
-        result = cls._simple_new(data, **d)
+        if not isinstance(data, DatetimeArray):
```

Comment:
what is all this about?

Reply:
backward compat for old pickles

Reply:
can you make this nicer. it is very hard to grok.

Reply:
pls comment the cases